### PR TITLE
Pr/clippy fixes

### PR DESCRIPTION
**Pull request by hunger**
Yet more clippy fixes... again pretty normal stuff by now.

### CODE REVIEW

File: sixtyfps_compiler/layout.rs [hunk 1]
```diff
@@ -227,14 +227,30 @@ impl LayoutConstraints {
...
 
     pub fn visit_named_references(&mut self, visitor: &mut impl FnMut(&mut NamedReference)) {
-        self.max_width.as_mut().map(|e| visitor(&mut *e));
-        self.min_width.as_mut().map(|e| visitor(&mut *e));
-        self.max_height.as_mut().map(|e| visitor(&mut *e));
-        self.min_height.as_mut().map(|e| visitor(&mut *e));
-        self.preferred_width.as_mut().map(|e| visitor(&mut *e));
-        self.preferred_height.as_mut().map(|e| visitor(&mut *e));
-        self.horizontal_stretch.as_mut().map(|e| visitor(&mut *e));
-        self.vertical_stretch.as_mut().map(|e| visitor(&mut *e));
```
Comment (ogoffart) [inline sixtyfps_compiler/layout.rs:237]: /me liked that code much better before

**Review by ogoffart — APPROVED** [review]
looks ok, although i really dislike that clippy don't like me to use Option::map that returns ()

If it was just me, i'd disable that lint globaly

File: sixtyfps_compiler/layout.rs (hunk 1 above)
Comment (hunger) [reply sixtyfps_compiler/layout.rs:237]: Yeap, that one indeed got more ugly!

I am actually surprised how little difference this makes in most places though! I had expected this particular lint to cause more ugliness in your code base (and I have never seen this `foo.map(||->() {...}` in any other project before.

**Comment by hunger** [comment]
So what about this PR? Should I drop the `map()` to `if let Some(...)` change and replace it with a global override?

I have never seen this trick in the wild, so it at least is not very common and might cause some users to wonder:-)

**Pull request merged**